src/stim/circuit/circuit_pybind_test.py

```diff
@@ -1345,3 +1345,32 @@ def test_num_ticks():
         CX 0 1
         TICK
     """).num_ticks == 2
+
+
+def test_reference_sample():
+    circuit = stim.Circuit(
+        """
+        H 0
+        CNOT 0 1
+    """
+    )
+    ref = circuit.reference_sample()
+    assert len(ref) == 0
+    circuit = stim.Circuit(
+        """
+        H 0
+        M 0
+        M 1
+    """
+    )
+    ref = circuit.reference_sample()
+    assert ref[0] == ref[1]
+    circuit = stim.Circuit()
+    circuit.append("X", (i for i in range(0, 100, 2)))
+    circuit.append("M", (i for i in range(100)))
+    ref_sample = circuit.reference_sample(bit_packed=False)
+    expected = np.zeros((len(ref_sample) // 8 + 1), dtype=np.uint8)
+    for bit_indx, bit in enumerate(ref_sample):
+        expected[bit_indx//8] |= (bit << (bit_indx % 8))
```

Comment:
Use `np.unpackbits` (little endian)  and `np.testing.assert_array_equal` for the comparison, instead of doing it custom here.